src/olympia/bandwagon/views.py

```diff
@@ -676,16 +674,27 @@ def get_queryset(self):
             author=self.get_account_viewset().get_object())
 
 
-class CollectionAddonViewSet(ListModelMixin, GenericViewSet):
-    permission_classes = [AllowAny]
+class CollectionAddonViewSet(ModelViewSet):
+    permission_classes = []  # We don't need extra permissions.
     serializer_class = CollectionAddonSerializer
+    lookup_field = 'addon'
 
-    def get_queryset(self):
-        if not hasattr(self, 'collection_object'):
-            self.collection_object = CollectionViewSet(
+    def get_collection_viewset(self):
+        if not hasattr(self, 'collection_viewset'):
+            # CollectionViewSet's permission_classes are good for us.
+            self.collection_viewset = CollectionViewSet(
                 request=self.request,
                 kwargs={'user_pk': self.kwargs['user_pk'],
-                        'slug': self.kwargs['collection_slug']}).get_object()
+                        'slug': self.kwargs['collection_slug']})
+        return self.collection_viewset
+
+    def get_object(self):
+        self.lookup_url_kwarg = self.lookup_url_kwarg or self.lookup_field
+        # if the lookup is not a number, its probably the slug instead.
+        if not self.kwargs[self.lookup_url_kwarg].isdigit():
```

Comment:
You don't test `self.kwargs[self.lookup_url_kwarg]` existence, or whether it's a string. Could it be absent/not a string ?